addons/binding/org.openhab.binding.icloud/README.md

```diff
@@ -0,0 +1,132 @@
+# iCloud Binding
+
+This binding uses the Apple iCloud service for providing device based information such as battery details or location.
+An Apple account is required. Two factor authentication is supported.
+
+Please note: Application specific passwords are not supported.
+
+The binding uses Google's "[Geocode API](https://developers.google.com/maps/documentation/geocoding/)" to translate the coordinates into a address data.
+
+<!-- TOC -->
+
+- [iCloud Binding](#icloud-binding)
+    - [Supported Things](#supported-things)
+    - [Discovery](#discovery)
+    - [Binding Configuration](#binding-configuration)
+    - [Thing Configuration](#thing-configuration)
+        - [iCloud Account Thing](#icloud-account-thing)
+    - [Channels](#channels)
+        - [Account](#account)
+        - [Device](#device)
+    - [Full Example](#full-example)
+        - [icloud.things](#icloudthings)
+        - [icloud.items](#iclouditems)
+- [Configuration with Paper UI](#configuration-with-paper-ui)
+
+<!-- /TOC -->
+
+## Supported Things
+
+The following devices are known to work with this binding:
+
+* iPhone 6s
+* iPhone 5c
+* iPhone 7
+* iWatch 2
+* iPad Air 2/2017/Pro
+* MacBook Pro
+
+Other devices should work as well. Please provide feedback if you have tested another device type. 
+
+## Discovery
+
+For each iCloud account to be included an iCloud binding thing needs to be configured with your id, password and refresh rate. The devices registered to this account will then be automatically discovered. Do not create devices manually.
+
+## Binding Configuration
+
+The binding has no configuration options, all configuration is done at Thing level.
+
+## Thing Configuration
+
+### iCloud Account Thing
+
+| Parameter            | Description                                         |           |
+|----------------------|-----------------------------------------------------|-----------|
+| appleId              |                                                     | mandatory |
+| password             |                                                     | mandatory |
+| refreshTimeInMinutes |                                                     | mandatory |
+| googleAPIKey         | Google API key to be used for address lookup calls. | optional  |
```

Comment:
Somewhere around here please describe how an device is identified (hash of device identifier) and what happens when it is disconnected from the cloud